src/TestImages.jl

```diff
@@ -46,53 +50,81 @@ remotefiles = [
+julia> testimage("cameraman") # without extension works
+julia> testimage("c") # with only partial name also works
 ```
+
+Use `TestImages.remotefiles` to get a full list of available images. You can also check
+https://testimages.juliaimages.org/
 """
-function testimage(filename, ops...)
-    isdir(imagedir) || mkpath(imagedir)
-    imagefile = joinpath(imagedir, filename)
-    if !isfile(imagefile)
-        fls = readdir(imagedir)
-        havefile = false
-        for f in fls
-            if startswith(f, filename)
-                imagefile = joinpath(imagedir, f)
-                havefile = true
-                break
-            end
-        end
+function testimage(filename; download_only = false, ops...)
+    imagefile = full_imagefile(filename)
 
-        if !havefile
-            @info "Could not find "*filename*" in directory images/ . Checking if it exists in the online repository."
-            for f in remotefiles
-                if startswith(f, filename)
-                    @info "Found "*filename*" in the online repository. Downloading to the images directory."
-                    download(REPO_URL*f*"?raw=true", joinpath(imagedir, f))
-                    havefile = true
-                    imagefile = joinpath(imagedir, f)
-                    break
-                end
-            end
-        end
-        havefile || throw(ArgumentError("$filename not found in the directory images/ or the online repository. use `TestImages.remotefiles` to get a full list of test images"))
-    end
-    img = load(imagefile, ops...)
-    if startswith(basename(imagefile), "mri-stack")
+    download_only && return imagefile
+
+    img = load(imagefile; ops...)
+    if basename(imagefile) == "mri-stack.tif"
         # orientation is posterior-right-superior,
         # see http://www.grahamwideman.com/gw/brain/orientation/orientterms.htm
         return AxisArray(img, (:P, :R, :S), (1, 1, 5))
     end
     img
 end
 
+"""get the first match of `filename` in remotefiles"""
+function full_imagename(filename)
+    idx = findfirst(remotefiles) do x
+        startswith(x, filename)
+    end
+    idx === nothing && throw(ArgumentError("$filename not found in the online repository, use `TestImages.remotefiles` to get a full list of test images."))
+
+    return remotefiles[idx]
+end
+
+# Pkg.Artifacts is introduce in julia v1.3.0
+if VERSION >= v"1.3.0"
+    function full_imagefile(filename)
+        imagename = full_imagename(filename)
+        file_hash = artifact_hash(imagename, artifacts_toml)
+
+        if file_hash === nothing || !artifact_exists(file_hash)
+            file_hash = create_artifact() do artifact_dir
+                @info "Found "*imagename*" in the online repository. Downloading to the artifact directory."
+                download(REPO_URL*imagename*"?raw=true", joinpath(artifact_dir, imagename))
+            end
+
+            bind_artifact!(artifacts_toml, imagename, file_hash)
+        end
```

Comment:
codes here follows the sample codes in https://julialang.github.io/Pkg.jl/dev/artifacts/#Using-Artifacts-1

However,  in `julia v1.3` on OSX and nightly built on Linux and OSX, 

```
LoadError: Mapping for 'cameraman.tif' within /home/travis/build/JuliaImages/TestImages.jl/Artifacts.toml already exists!
```

is observed here since `create_artifact`  generates a new hash, which differs from the one in `Artifact.toml`.